File: simulator/demo/src/main/java/com/example/FraudDetectorService1.java
package com.example;

import org.apache.kafka.clients.consumer.ConsumerRecord;

public class FraudDetectorService1 {
  private final String topic = "ECOMMERCE_NEW_ORDER";

  public static void main(String[] args) {
    var fraudDetectorService1 = new FraudDetectorService1();
    try (var service = new KafkaService(FraudDetectorService1.class.getSimpleName(), fraudDetectorService1.topic,
        fraudDetectorService1::parser)) {
      service.run();
    }
  }

  private void parser(ConsumerRecord<String, String> record) {
    System.out.println("-----------------------------------------");
    System.out.println("Processing new order, checking for fraud");
    System.out.println("Key " + record.key());
    System.out.println("Value " + record.value());
    System.out.println("Partition " + record.partition());
    System.out.println("Offset " + record.offset());

    try {
      Thread.sleep(500);
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
    System.out.println("Order proccessed");
  }
}

File: simulator/project_python/fraud_detector_service1.py
from kafka_service import KafkaService


class FraudDetectorService:
    topics = "ECOMMERCE_NEW_ORDER"

    def __init__(self):
        service = KafkaService(FraudDetectorService.__name__, self.topics, self.parser)
        try:
            service.run()
        except Exception as e:
            service.close(e)

    def parser(self, record):
        print("-----------------------------------------")
        print("Processing new order, checking for fraud")
        print("Key => {}".format(record.key.decode("utf-8")))
        print("Value => {}".format(record.value.decode("utf-8")))
        print("Partition => {}".format(record.partition))
        print("Offset => {}".format(record.offset))
        print("Order proccessed\n")


FraudDetectorService()
